src/Core/Localization/DataLayer/LocaleDataLayerInterface.php

```diff
@@ -74,5 +74,5 @@ public function write($localeCode, $localeData);
      *
      * @return self
      */
-    public function setLowerLayer(LocaleDataLayerInterface $lowerLayer);
+    public function setLowerLayer(self $lowerLayer);
```

Comment:
As this is an interface I think it's better to keep the explicit Interface in the prototype